content/updates/understanding-the-terraform-check-block-feature.md

```diff
@@ -0,0 +1,210 @@
+
+First, checks can work great when the actual resource provisioning isn't the primary focus of the assertion. Given the necessity of configuring sandbox environments (whether ephemeral or persistent, hosted in the cloud, or run locally using mocking tools) and writing tests in Ruby or Go, native HCL `check` blocks can be seen as a practical alternative.
+
+Furthermore, Terraform uses declarative language to define your infrastructure's desired state and, by its nature, does not specify the steps needed to reach this state. Examining the details of the expected state to confirm proper changes while writing tests can quickly become tedious. An effective test could be to check if the changes have indeed been applied - for this, you only need to look at one part of the result. For example, confirm that the worker nodes have joined the EKS cluster.
+
+So give it a try, assess your current validation and testing kit and determine scenarios where checks could be beneficial, either as an addition or a replacement.
+
+## How can this be set up?
+
+Adding a simple check is trivial. Let’s assume we have a root module that manages AWS EKS cluster configuration and contains the resource `aws_eks_cluster`:
+```hcl
+resource "aws_eks_cluster" "default" {
+  name     = "my-cluster"
+  role_arn = var.role_arn
+  vpc_config {
+    subnet_ids = var.subnet_ids
+  }
+}
+```
+
+To start using the `check` block, ensure you’re using Terraform v1.5+. Since the release of terraform 1.0, we recommend to use the [pessimistic constraint operator](https://developer.hashicorp.com/terraform/language/expressions/version-constraints#-3) (i.e. `~>`) to set bounds on the version of Terraform and the providers a [root module](https://developer.hashicorp.com/terraform/language/modules#the-root-module) depends on:
+
+```hcl
+terraform {
+  required_version = "~> 1.5"
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.0"
+    }
+  }
+}
+```
+
+Let’s define a check that validates that our cluster instance is healthy and available:
+```hcl
+check "aws_eks_cluster_default" {
+  assert {
+    condition     = aws_eks_cluster.default.status == "ACTIVE"
+    error_message = "EKS cluster ${aws_eks_cluster.default.id} status is ${aws_eks_cluster.default.status}"
+  }
+}
+```
+
+## Using a data source in the assertions
+
+In addition to simple assertions, Terraform offers the ability to reference a data source within `check` block assertions. It’s queried by Terraform at the end of each plan and apply operations to evaluate the checks and obtain the most recent data from your environment.
+
+Terraform’s operation won’t be interrupted by failures in the scoped data block or any unsuccessful assertions. This unlocks continuous validation of your assumptions about the infrastructure rather than being confined to the point of initial provisioning.
```

Comment:
```suggestion
**Terraform’s operation won’t be interrupted by failures in the scoped data block or any unsuccessful assertions**. This unlocks continuous validation of your assumptions about the infrastructure rather than being confined to the point of initial provisioning.
```